Add Configs CI Infrastructure (#18)
ACCESS-NRI/access-esm1.5-configs: CONTRIBUTING.md

5. Optionally, you can now modify your PR and get more reproducibility checks. Particularly in the case where bitwise reproducibility should be retained this is an opportunity to modify the configuration to enable this.
6. Bump the version using the `!bump [major|minor]` command depending on the result of the reproducibility check. Additionally, if the checksums are different, the updated checksum will be automatically committed to the PR. Bumping the version in some way is a requirement before the PR will be mergable.
7. Merge the PR

### Common Changes Required

#### Required metadata

Each configuration should have a metadata file, `metadata.yaml`, which uses the ACCESS-NRI [experiment schema](https://github.com/ACCESS-NRI/schema/blob/main/au.org.access-nri/model/output/experiment-metadata/README.md). Note that required fields of the schema are removed from file validations checks as experiment specific fields such as UUIDs should not be set for released configurations. The following fields must be set in `metadata.yaml`:

##### version

Use the existing `release-*` version. If there isn't an existing version set to `null`.

##### realm

> Acceptable values for this field must be set as a sequence:

Reviewer: ```suggestion
Acceptable values for this field are shown below and must be set as a sequence:
```

```yaml

> realm:
>     - <acceptable value 1>
>     - <acceptable value 2>
>     # - ...

Reviewer: ```suggestion
realm:
    - atmosphere
    - ocean
    - seaIce
    - land
    - ocnBgchm
```

```

##### nominal resolution

> Choose the appropriate value for the resolution used:

Reviewer: ```suggestion
There is currently only one supported nominal resolution:
```

| Config resolution | Nominal Resolution |
| -- | -- |
| 1&deg; | 100 km |

> `<Add more as applicable>`

Reviewer: ```suggestion
```

These are sourced from [the CMIP6 controlled vocabulary](https://github.com/WCRP-CMIP/CMIP6_CVs/blob/main/CMIP6_nominal_resolution.json). If your resolution differs from those listed you will need to make a pull request to add it to this documentation and the QA checks.

##### keywords

We have a "controlled vocabulary of keywords to prevent a proliferation of synonyms that mean the same thing, and to make it easy to populate these fields:

| Topic | Keywords (mutually exclusive) |
|--|--|

> | `<topic>` | `<keywords>` |

Reviewer: ```suggestion
| Spatial extent | `global`|
| Model | `access-esm1.5` |
```

I am sure there are other keywords we could suggest, but I'm not familiar enough with ESM to know. Will add to the ESM stand-up to ask.

##### reference

> An appropriate scientific reference for the configuration. `<Give example of an appropriate scientific reference for this model>`

Reviewer: ```suggestion
An appropriate scientific reference for the configuration. For ACCESS-ESM1.5 this should be https://doi.org/10.1071/ES19035 if there is no more appropriate reference.
```

##### license

This is the license that will apply to the model outputs for an experiment. This should be set to the [SPDX identifier](https://spdx.org/licenses/) for [CC BY 4.0](https://creativecommons.org/licenses/by/4.0/) (`CC-BY-4.0`) to alleviate users from the burden of choosing a license, and to ensure model outputs have a permissive license for reuse to encourage open and shareable science.

##### url

This is a bit tricky. Ideally this should be a URL to the GitHub (or similar) repository of the configuration *for the experiment being run*. So if we include this and require it to be filled then it should either be the URL pointing at the branch being modified, or a placeholder. Either way it should include a comment that it should be updated to reference the experiment being run.

##### model

> `<Give acceptable model names>`

Reviewer: ```suggestion
Should be `access-esm1.5`
```

#### Configuration settings

> ##### restart_period
> 
> This is checked to make sure a shorter run time hasn't been set during testing and forgotten to set back to the proper value. As it is difficult to create a general heuristic the values have been hard-coded to those shown below:
> 
> | Config resolution | `restart_period`|
> | -- | -- |
> 
> `<This table needs to be filled in for the model in question>`

Reviewer: ```suggestion
##### calendar

The values in `calendar` are checked to make sure the correct `start` and `runtime` are set. 

`runtime` should always be `year:1`, `months:0`, `days:0` because land-use changes are recalculated every year for the `historical` configuration. This is not the case for `preindustrial`, but it is kept at 1 year for consistency.

As it is difficult to create a general heuristic the values for `start` have been hard-coded to those shown below:

| Config | `year`| `month` | `day`
| preindustrial | 101 | 1 | 1 |
| historical | 1850 | 1 | 1 |
```

##### restart_freq

This governs how what model restart files are retained.

> The requirement is simply that a date-based frequency be used so that restarts are saved in a reliable manner. Typical values are `1YS` or `5YS` for 0.1&deg; models, and `5YS` to `20YS` for 1&deg;. See the [payu documentation](https://payu.readthedocs.io/en/latest/config.html#model) for a detailed description of the meaning of the time units in `restart_freq`.

Reviewer: ```suggestion
The requirement is simply that a date-based frequency be used so that restarts are saved in a reliable manner. A typical value is `20YS`. See the [payu documentation](https://payu.readthedocs.io/en/latest/config.html#model) for a detailed description of the meaning of the time units in `restart_freq`.
```
